3p/integration.js

```diff
@@ -143,8 +143,25 @@ const AMP_EMBED_ALLOWED = {
   zergnet: true,
 };
 
-const data = parseFragment(location.hash);
-window.context = data._context;
+let data;
+// Need to cache iframeName as it will be potentially overwritten by
+// masterSelection, as per below.
+const iframeName = window.name;
+try {
```

Comment:
can you move the try-catch into `parseFragment`? And make it return `null` for exception.